pkg/server/httpAPI.go

```diff
@@ -233,15 +232,7 @@ func (h *HTTPHandler) serveInitialize(rw http.ResponseWriter, req *http.Request)
 		go func() {
 			var mode validator.Mode
 
-			// This is needed to stop the currently running snapshotter.
-			if h.Snapshotter != nil {
-				h.SetStatus(http.StatusServiceUnavailable)
-				atomic.StoreUint32(&h.AckState, HandlerAckWaiting)
-				h.Logger.Info("Changed handler state.")
-				h.ReqCh <- emptyStruct
-				h.Logger.Info("Waiting for acknowledgment...")
-				<-h.AckCh
-			}
+			h.SetStatus(http.StatusServiceUnavailable)
```

Comment:
Shouldn't the status be set to `Unavailable` only when `h.Snapshotter` is present?

Reply:
it doesn't matter.
For single node case: we know `h.Snapshotter!=nil` is always true as it is the etcd-leader.
For multi-node: we are not using the `/healthz` endpoint as readinessProbe, so it is not used anywhere.